### DISCUSSION TITLE
how to use stateNotifierProvider or changeNotifierProvider that depends on future Provider

### DISCUSSION THREAD

**Discussion by gopiKrishnaPuligundla** [rrousselGit/riverpod#1547] [General]
Hi All,

my requirement is , i want to have a listview that displays list of quotes which are stored in object box .
i tried changenotifierprovider to keep track of CRUD for quotes , so that listview widget gets rebuild.
but changenotifierProvider initialisation needs a object box instance which is a future , so i defined a future provider for initialising object box.
and in changenotifierprovider initialisation am listening to future provider as below, but it is not working .. please find code snippet below

final objectBoxProvider = FutureProvider<ObjectBox>((ref) async {
    return await ObjectBox.init();
});
//final quotesAllProvider = StateNotifierProvider<AsyncValue<List<Quote>>>((ref) async {
final OBNProvider = ChangeNotifierProvider<ObjectBoxNotifier>((ref) {
    ObjectBox? ob = ref.read(objectBoxProvider).value;
    //return ob?.getAllQuotes();
    return ObjectBoxNotifier(objectBox: ob);
} );
i feel some basic understanding of riverpod is wrong from my end.
git repo link  : https://github.com/gopiKrishnaPuligundla/japa_counter/blob/dev_branch/

**Comment by gopiKrishnaPuligundla** [comment]
any inputs would be helpful .. am new to flutter

**Comment by Clashkid155** [comment]
I don't really understand the question but I think this is what you want.
```dart
final objectBoxProvider = FutureProvider((ref) async {
        return await ObjectBox.init();
});
//final quotesAllProvider = StateNotifierProvider<AsyncValue<List>>((ref) async {
final OBNProvider = ChangeNotifierProvider((ref) {
       ObjectBox? ob = ref.watch(objectBoxProvider).value;
      //return ob?.getAllQuotes();
      return ObjectBoxNotifier(objectBox: ob);
} );
```

**Reply by gopiKrishnaPuligundla** [reply]
Hi Clashkid155 , thanks for reply , your answer helps , one question is, if we want to use StateNotifierProvider for above use case,  if a table content changes in objectBox we have to get the new state , how to implement that ?i mean objectBox reference wont change , only the table content changes.
